dotnet/src/Connectors/Connectors.Onnx/OnnxRuntimeGenAIPromptExecutionSettings.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Text.Json.Serialization;
+
+namespace Microsoft.SemanticKernel.Connectors.Onnx;
+
+/// <summary>
+/// OnnxRuntimeGenAI Execution Settings.
+/// </summary>
+public sealed class OnnxRuntimeGenAIPromptExecutionSettings : PromptExecutionSettings
+{
+    /// <summary>
+    /// Convert PromptExecutionSettings to OnnxRuntimeGenAIPromptExecutionSettings
+    /// </summary>
+    /// <param name="executionSettings"></param>
+    /// <returns></returns>
+    public static OnnxRuntimeGenAIPromptExecutionSettings FromExecutionSettings(PromptExecutionSettings? executionSettings)
+    {
+        switch (executionSettings)
+        {
+            case OnnxRuntimeGenAIPromptExecutionSettings settings:
+                return settings;
+            default:
+                return new OnnxRuntimeGenAIPromptExecutionSettings();
```

Comment:
We should try to create the settings from what's passed in. The other connectors do this be serializing the input settings and then deserializing as the desired settings type.